AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -537,6 +541,7 @@ - (void)collectionView:(UICollectionView *)collectionView willDisplayCell:(UICol
   if (cellNode.neverShowPlaceholders) {
     [cellNode recursivelyEnsureDisplaySynchronously:YES];
   }
+  [_cellsForVisibilityUpdates addObject:cell];
```

Comment:
This is where we should check if [cell node] respondsToSelector: / needs the visibility calls.

You can change the type of the argument to willDisplayCell:(_ASCollectionViewCell *)cell  - so that you can call the -node method on it.

You should still add the cell itself to the set, but check the node for whether it needs the notifications.